GNUSTEP_MAKEFILES needs to be set earlier

When building without `GNUSTEP_MAKEFILES` already set in the environment, the configure script shipped in `gnustep-back-0.31.0.tar.gz` fails immediately with:

```
configure: error: cannot find required auxiliary files:
```

There is code to set `GNUSTEP_MAKEFILES` automatically, but it doesn't run until after the generated configure script has already tried to use it to find config.guess and config.sub; it probably needs moving earlier in the file.

--- a/configure
+++ b/configure
@@ -2562,7 +2562,11 @@ as_fn_append ac_header_c_list " unistd.h unistd_h HAVE_UNISTD_H"
 ac_aux_files="config.guess config.sub"
 
 # Locations in which to look for auxiliary files.
-ac_aux_dir_candidates="$GNUSTEP_MAKEFILES"
+ac_aux_dir_candidates="`if test -z "$GNUSTEP_MAKEFILES"; then
+  gnustep-config --variable=GNUSTEP_MAKEFILES 2>&5
+else
+  echo $GNUSTEP_MAKEFILES
+fi`"
 
 # Search for a directory containing all of the required auxiliary files,
 # $ac_aux_files, from the $PATH-style list $ac_aux_dir_candidates.

--- a/configure.ac
+++ b/configure.ac
@@ -41,8 +41,15 @@ fi
 
 #--------------------------------------------------------------------
 # Use config.guess, config.sub and install-sh provided by gnustep-make
+# Evaluation order of the configure script means we can not depend on
+# GNUSTEP_MAKEFILES being set for simple variable substitution, so we
+# execute a bit of shell code to dynamically get the directory.
 #--------------------------------------------------------------------
-AC_CONFIG_AUX_DIR($GNUSTEP_MAKEFILES)
+AC_CONFIG_AUX_DIR([`if test -z "$GNUSTEP_MAKEFILES"; then
+  gnustep-config --variable=GNUSTEP_MAKEFILES 2>&5
+else
+  echo $GNUSTEP_MAKEFILES
+fi`])
 
 #--------------------------------------------------------------------
 # Determine the host, build, and target systems

